crypto/bls12381/aggregation.go

```diff
@@ -0,0 +1,47 @@
+//go:build bls12381
+
+package bls12381
+
+import (
+	"errors"
+
+	blst "github.com/supranational/blst/bindings/go"
+)
+
+// ErrAggregation is returned when aggregation fails.
+var ErrAggregation = errors.New("bls12381: failed to aggregate signatures")
+
+// For minimal-pubkey-size operations.
+//
+// Changing this to 'minimal-signature-size' would render CometBFT not Ethereum
+// compatible.
+type (
+	blstAggregateSignature = blst.P2Aggregate
+)
+
+// AggregateSignatures aggregates the given compressed signatures.
+//
+// Does not group-check the signatures.
+func AggregateSignatures(sigsToAgg [][]byte) ([]byte, error) {
+	var agProj blstAggregateSignature
+	if !agProj.AggregateCompressed(sigsToAgg, false) {
```

Comment:
You're passing `false` here because we know we have already verified each signature when received from the network. Is that accurate?

Reply:
Yes.